config/config_partial.go

```diff
@@ -99,6 +99,16 @@ func DecodeBaseBlocks(
 	filename string,
 	includeFromChild *IncludeConfig,
 ) (*cty.Value, *terragruntInclude, TrackInclude, error) {
+	// first decode of IamRole which can be referenced in other blocks
+	contextExtensions := EvalContextExtensions{}
+	flags := terragruntFlags{}
+	err := decodeHcl(hclFile, filename, &flags, terragruntOptions, contextExtensions)
+	if err != nil {
+		return nil, nil, TrackInclude{}, err
+	}
```

Comment:
Doing this parsing here means that we won't honor `iam_role` set on parent configurations that are `include`d. You'll need to implement a special version of `PartialParseConfigFile` that bypasses locals and only parses `include` + `iam_role` to handle this.

Reply:
Make sense, will look in this dirrection